### DISCUSSION TITLE
Vue 3.4 defineModel() Macro supported?

### DISCUSSION THREAD

**Discussion by db-lyon** [quasarframework/quasar#16875] [General - Components / Directives / etc] [answered]
I'm several months into a vue-cli project and just discovered Quasar. 

I use the new [defineModel() macro](https://vuejs.org/guide/components/v-model.html) extensively. wondering if Quasar components already support it, or if that's something I would need to contribute.

**Comment by yusufkandemir** [comment, marked as answer]
It's not a special thing, it would work with any Vue component that has a `v-model`. So, you can use it with Quasar UI components. Quasar CLI-wise, it's supported as well.
P.S. I think it's easier to test this than to ask, but anyway.

**Comment by angelhdzmultimedia** [comment]
The statements

> It's not a special thing

and

> P.S. I think it's easier to test this than to ask, but anyway.

 should be revisited.

 `defineModel` macro was hidden behind an experimental flag until recently.
So the OP question and concern is valid. 

In Nuxt we had to enable it in the config, behind the vue options.

It's already out of experimental, so it will currently work with Quasar out of the box while Quasar keeps the bumping of vue package version consistent.

I'm very concerned with the constant attitude and answers from some of the members of the Quasar team. 
Not inspiring trust, kindness or "team effort".  It sometimes discourages me to open issues, ask questions, continue using this framework or contributing to it. 😔

But I still wish everyone health, peace, and success regardless. 👋🔥💙💚

**Reply by yusufkandemir** [reply]
He explicitly stated Vue 3.4, which shipped `defineModel` as stable. Furthermore, he is not asking if he can use `defineModel` in a Quasar CLI project, he is asking whether he can use it with Quasar components. That's why I stated it's not a special thing, as you can do `const model = defineModel()` then you can use `model` with any component, whether custom-made or from a UI library. In case he would use Quasar CLI, I also stated the CLI supports it too. And, later, I added it would be easier to test this than ask, as that would've saved everyone's time. We are helping a lot of people, and it would be better if people with genuine problems get help, rather than people asking theoretical questions which take more time to ask than to test it. It might be the first time this person did this, and they might not have been experienced in asking questions, they might not have been realized, etc. So, a simple reminder could make them rethink their choice and approach it differently the next time they ask a question. You or someone else could perceive this as a bad attitude, which is sad, but this is our best effort while doing our daily jobs, some extra jobs, and contributing by code/helping/triaging. It certainly could've been done in a much nicer way, but we can't be perfect all the time. Thanks for understanding.

**Reply by angelhdzmultimedia** [reply]
Ok, I offer an apology. I didn't notice the explicit version, that's on me.
I was scrolling through the discussions to see if I could be of help and saw this, and my first impression was the one I expressed, also because I was recently searching info about if `propsDestructuring` was also removed from experimental as `defineModel` and didn't find anything and got frustrated. 

As always, thanks to the maintainers for all the hard work, and I also want to contribute more. I suffer from generalized anxiety, and it's hard for me to focus, and not to get driven by the overthinking. 

Best wishes, cheers. 👋😊

**Reply by yusufkandemir** [reply]
> I was recently searching info about if propsDestructuring was also removed from experimental as defineModel and didn't find anything and got frustrated.

I believe you mean the reactive props destructure feature of Reactivity Transform. The Reactivity Transform has been officially dropped, see [the warning in Vue docs](https://vuejs.org/guide/extras/reactivity-transform#reactivity-transform). However, the reactive props destructing has been made a separate feature, and you might already be aware of this. But, it seems that this has not been documented well. Anyways, the source code confirms it's still experimental: https://github.com/vuejs/core/blob/3f92126a26a544dfa69211cf2977556a2706bb2c/packages/compiler-sfc/src/script/definePropsDestructure.ts#L35-L40

As for the other things, no problem, and thanks. I hope you can get over those problems and have less stressful days.
